client/components/survey-container/components/question-single-selection.tsx

```diff
@@ -1,17 +1,56 @@
-import { RadioControl } from '@wordpress/components';
 import { QuestionSelectionType } from './question-step';
 
 const QuestionSingleSelection = ( { onChange, question, value }: QuestionSelectionType ) => {
-	// Placeholder component
 	return (
-		<div className="question__single-selection">
-			<RadioControl
-				options={ question.options }
-				selected={ value[ 0 ] }
-				onChange={ ( value ) => {
-					onChange( question.key, [ value ] );
-				} }
-			/>
+		<div className="question-options__container" role="radiogroup">
+			{ question.options.map( ( option, index ) => {
+				const isSelected = value.includes( option.value );
+				const handleKeyDown = ( event: React.KeyboardEvent< HTMLDivElement > ) => {
+					if ( event.key === 'Enter' || event.key === ' ' ) {
+						onChange( question.key, [ option.value ] );
+					}
+				};
+
+				return (
```

Comment:
This inner part could be extracted into a new component to avoid creating variables inside the markup.
```
{ question.options.map( ( option, index ) => {
   <__RadioOption option={ option } ...>
} }
```

Reply:
Thanks, Paulo! Would you also use that `__RadioOption` name for it? 🤔

Reply:
No, sorry I wasn't clear, it's just a placeholder. I don't have a name for it, but it could be something like `SurveyRadioOption`. And for consistency, the main component could be renamed from `QuestionSingleSelection` to `SurveyRadioControl`.

Reply:
This is now resolved.